src/EventSubscriber/RegistrationSubscriber.php

```diff
@@ -71,13 +73,18 @@ public function persistRegistrationAdded(RegistrationAddedEvent $event): void
 
     public function persistRegistrationRemoved(RegistrationRemovedEvent $event): void
     {
-        $event->getRegistration()->setDeleteDate(new \DateTime('now'));
+        if ($event->getRegistration()->getPerson() instanceof ExternalRegistrant) {
```

Comment:
We're calling `getRegistration` thrice here. maybe add somthing above this line akin to
```php
$registration = $event->getRegistration()
```